docs/FeatureManagement/Clients/Microsoft.FeatureManagement.AspNetCore.md

```diff
@@ -0,0 +1,367 @@
+
+```
+{ 
+  "Logging": {
+    "LogLevel": {
+      "Default": "Warning"
+    }
+  },
+
+  // Define feature flags in a json file
+  "FeatureManagement": {
+    "FeatureT": {
+      "EnabledFor": [
+        {
+          "Name": "AlwaysOn"
+        }
+      ]
+    }
+    "FeatureU": {
+      "EnabledFor": []
+    }
+    "FeatureV": {
+      "EnabledFor": [
+        {
+          "Name": "TimeWindow",
+          "Parameters": {
+            "Start": "Wed, 01 May 2019 13:59:59 GMT",
+            "End": "Mon, 01 July 2019 00:00:00 GMT"
+          }
+        }
+      ]
+    }
+  }
+}
+```
+
+The `FeatureManagement` section of the json document is used by convention to load feature flag settings. In the section above, we see that we have provided three different features. Features define their feature filters using the `EnabledFor` property. In the feature filters for `FeatureT` we see `AlwaysOn`. This feature filter is built-in and if specified will always enable the feature. The `AlwaysOn` feature filter does not require any configuration so it only has the _Name_ property. `FeatureU` has no filters in its `EnabledFor` property and thus will never be enabled. Any functionality that relies on this feature being enabled will not be accessible as long as the feature filters remain empty. However, as soon as a feature filter is added that enables the feature it can begin working. `FeatureV` specifies a feature filter named `TimeWindow`. This is an example of a configurable feature filter. We can see in the example that the filter has a parameter's property. This is used to configure the filter. In this case, the start and end times for the feature to be active are configured.
+
+### On/Off Declaration
+ 
+The following snippet demonstrates an alternative way to define a feature that can be used for on/off features. 
+```
+{
+  "Logging": {
+    "LogLevel": {
+      "Default": "Warning"
+    }
+  },
+
+  // Define feature flags in config file
+  "FeatureManagement": {
+    "FeatureT": true, // On feature
+    "FeatureX": false // Off feature
+  }
+}
+```
+### Referencing
+
+To make it easier to reference these feature flags in code, we recommend to define feature flag variables like below.
+
+    // Define feature flags in an enum
+    [FeatureFlag]
+    public enum MyFeatureFlags
+    {
+        FeatureT,
+        FeatureU,
+        FeatureV
+    }
+    
+### Service Registration
+
+Feature flags rely on .NET Core dependency injection. We can register the feature management services using standard conventions.
+
+```
+using Microsoft.FeatureManagement;
+using Microsoft.FeatureManagement.FeatureFilters;
+
+public class Startup
+{
+  public void ConfigureServices(IServiceCollection services)
+  {
+      services.AddFeatureManagement()
+              .AddFeatureFilter<BrowserFilter>()
+              .AddFeatureFilter<SuperUserFilter>();
```

Comment:
Should we use the 2 built-in filters as examples? It will increase the chances that people copy the code and it just works.

Reply:
Good idea. Done.